### ISSUE TITLE
Undefined index in CacheKey::getInAndNotInClauses()

### ISSUE THREAD

**Issue by mycarrysun** [GeneaLabs/laravel-model-caching#227]
**Describe the bug**
Getting an undefined index error inside the `getInAndNotInClauses` function. [This line](https://github.com/GeneaLabs/laravel-model-caching/blob/eeb78e0a2b0dc1ffdfe2d2669c5d3bc3d2274c6b/src/CacheKey.php#L218).

**Eloquent Query**
Please provide the complete eloquent query that caused the bug, for example:
```php
$permit_type_ids = $request->input('permit_type_ids', []); //here this equals just []
Stat::where( [
	'statable_type' => 'permit_type',
	'slug'          => '30_day_avg_inv',
] )
    ->whereDate( 'created_at', '>', Carbon::now( TIMEZONE )->subDays( 30 ) )
    ->whereIn( 'statable_id', $permit_type_ids )
    ->oldest( 'created_at' )
    ->get();
```

**Stack Trace**
```
#0 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CacheKey.php(218): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'Undefined offse...', '/home/mike/Phps...', 218, Array)
#1 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CacheKey.php(173): GeneaLabs\\LaravelModelCaching\\CacheKey->getInAndNotInClauses(Array)
#2 [internal function]: GeneaLabs\\LaravelModelCaching\\CacheKey->GeneaLabs\\LaravelModelCaching\\{closure}('-nested-statabl...', Array)
#3 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1445): array_reduce(Array, Object(Closure), NULL)
#4 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CacheKey.php(177): Illuminate\\Support\\Collection->reduce(Object(Closure))
#5 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CacheKey.php(39): GeneaLabs\\LaravelModelCaching\\CacheKey->getWhereClauses()
#6 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/Traits/Caching.php(76): GeneaLabs\\LaravelModelCaching\\CacheKey->make(Array, NULL, '')
#7 /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CachedBuilder.php(98): GeneaLabs\\LaravelModelCaching\\CachedBuilder->makeCacheKey(Array)
#8 /home/mike/PhpstormProjects/CustomPermitsApp/api/app/Api/Controllers/AnalyticsController.php(82): GeneaLabs\\LaravelModelCaching\\CachedBuilder->get()
```

**Environment**
 - PHP: 7.2
 - OS: Ubuntu 18.10
 - Laravel: 5.8.8
 - Model Caching: 0.4.11

**Additional context**
Exception:
```
Undefined offset: 3 at /home/mike/PhpstormProjects/CustomPermitsApp/api/vendor/genealabs/laravel-model-caching/src/CacheKey.php:218)
```

**Comment by mikebronner** [comment]
@mycarrysun Thanks for reporting this. Will look into this soon. If you want to take a stab at it, my guess would be that I am not checking for empty arrays in `getInAndNotInClauses()`.

**Comment by mycarrysun** [comment]
Got it @mikebronner will submit a PR. Just going to check if `isset()` first

**Comment by mikebronner** [comment]
@mycarrysun This should now be fixed in release 0.4.12. Let me know if it works. :)
